### PR TITLE
service config: move balancer config parsing to internal

### PR DESCRIPTION
**Pull request by menghanl**
And make it a json unmarshaller, easy to use.

### CODE REVIEW

File: internal/serviceconfig/serviceconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+type testBalancerBuilder struct{}
+
+func (testBalancerBuilder) ParseConfig(js json.RawMessage) (externalserviceconfig.LoadBalancingConfig, error) {
+	if string(js) != testBalancerConfigJson {
+		return nil, fmt.Errorf("unexpected config json")
+	}
+	return testBalancerConfig, nil
+}
+
+func (testBalancerBuilder) Build(cc balancer.ClientConn, opts balancer.BuildOptions) balancer.Balancer {
```
Comment (dfawley) [inline internal/serviceconfig/serviceconfig_test.go:52]: Nit: embed balancer.Builder instead.

File: internal/serviceconfig/serviceconfig_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		wantErr bool
+	}{
+		{
+			name:    "empty json",
+			json:    "",
+			wantErr: true,
+		},
+		{
+			name:    "more than one entry for a name",
+			json:    `[{"balancer1":"1","balancer2":"2"}]`,
```
Comment (dfawley) [inline internal/serviceconfig/serviceconfig_test.go:78]: should these both be "balancer1" then?

File: internal/serviceconfig/serviceconfig_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			json: fmt.Sprintf(`[{"balancer1":"1"},{%q: %v}]`, testBalancerBuilderName, testBalancerConfigJson),
+			want: BalancerConfig{
+				Name:   testBalancerBuilderName,
+				Config: testBalancerConfig,
+			},
+			wantErr: false,
+		},
+		{
+			name: "balancer registered but builder not parser",
+			json: fmt.Sprintf("[{%q: %v}]", roundrobin.Name, testBalancerConfigJson),
```
Comment (dfawley) [inline internal/serviceconfig/serviceconfig_test.go:106]: Using RR for this test seems potentially bad. if we ever added service config parsing support to RR, this test will fail.

File: internal/serviceconfig/serviceconfig_test.go (hunk 1 above)
Comment (menghanl) [reply internal/serviceconfig/serviceconfig_test.go]: Done

File: internal/serviceconfig/serviceconfig_test.go (hunk 2 above)
Comment (menghanl) [reply internal/serviceconfig/serviceconfig_test.go:88]: This is a map with two entries, we want one entry.
Changed name and added comment.

File: internal/serviceconfig/serviceconfig_test.go (hunk 3 above)
Comment (menghanl) [reply internal/serviceconfig/serviceconfig_test.go]: Done

**Pull request merged**